### PR TITLE
build: fix Makefile ldflags for mac and mobile, bump gomobile deps, enable gomodules

### PR DESCRIPTION
**Pull request by djkazic**
## Change Description
PR [6963](https://github.com/lightningnetwork/lnd/pull/6963) refactored the Makefile to have a more streamlined approach to ldflags. However, in the process it also introduced two subtle but critical errors.
1. The `-ldflags` parameter is missing.
2. The ldflags themselves are not wrapped in a string.

The result is that the arguments are mangled, and gomobile complains about `-s` not being a parameter.

Before 6963:
```GO111MODULE=off /home/synthlock/go/bin/gomobile bind -target=android -tags="mobile dev autopilotrpc chainrpc invoicesrpc neutrinorpc peersrpc routerrpc signrpc verrpc walletrpc watchtowerrpc wtclientrpc" -ldflags " -s -w -X github.com/lightningnetwork/lnd/build.Commit=v0.15.5-beta -X github.com/lightningnetwork/lnd/build.CommitHash=c0a09209782b1c62c3393fcea0844e095c25046b -X github.com/lightningnetwork/lnd/build.GoVersion=go1.18.10 -X github.com/lightningnetwork/lnd/build.RawTags=" -v -o /home/synthlock/go/src/github.com/lightningnetwork/lnd/mobile/build/android/Lndmobile.aar github.com/lightningnetwork/lnd/mobile```
After 6963:
```GO111MODULE=off /home/synthlock/go/bin/gomobile bind -target=android -tags="mobile dev autopilotrpc chainrpc invoicesrpc neutrinorpc peersrpc routerrpc signrpc verrpc walletrpc watchtowerrpc wtclientrpc"  -s -w -buildid= -X github.com/lightningnetwork/lnd/build.Commit=kvdb/v1.4.1-6-g9187d4679 -v -o /home/synthlock/go/src/github.com/lightningnetwork/lnd/mobile/build/android/Lndmobile.aar github.com/lightningnetwork/lnd/mobile```

This PR adds in the correct behavior for ldflag passing. `-ldflags` is explicitly set in each affected stanza, and the ldflags are wrapped in a string.

## Steps to Test
`make android` on v0.15.5 release versus master.

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [x] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [x] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [x] Any new logging statements use an appropriate subsystem and logging level.
- [x]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

**Review by guggero — CHANGES_REQUESTED** [review]
Thanks for the fix! I can confirm this fixes at least one part of the issues.
It seems like we need some additional things to fix the mobile build. Not sure what exactly changed, but the same changes were necessary to get LNC compiling (see https://github.com/lightninglabs/lightning-node-connect/pull/66).

Here's the diff that enabled me to get Android compilation working. Would be great if you could test iOS as I don't have the Xcode framework installed.

[mobile.diff.txt](https://github.com/lightningnetwork/lnd/files/10508685/mobile.diff.txt)

**Comment by djkazic** [comment]
@guggero Interesting. Does this not build for you? I can apply the patch, but it seems like the main changes there are bumping the gomobile and bind deps.

**Comment by djkazic** [comment]
Commit messages edited and signed.

File: docs/release-notes/release-notes-0.16.0.md [hunk 1]
```diff
@@ -137,6 +137,9 @@ https://github.com/lightningnetwork/lnd/pull/6963/)
 * [A wire parsing bug has been fixed that would cause lnd to be unable _decode_
 certain large transactions](https://github.com/lightningnetwork/lnd/pull/7100).
 
+* [ldflags were being incorrectly passed](
```
Comment (guggero) [inline docs/release-notes/release-notes-0.16.0.md:140]: nit: "... in mobile builds"?

File: Makefile [hunk 2]
```diff
@@ -10,7 +10,7 @@ GOIMPORTS_PKG := github.com/rinchsan/gosimports/cmd/gosimports
 GO_BIN := ${GOPATH}/bin
 BTCD_BIN := $(GO_BIN)/btcd
 GOIMPORTS_BIN := $(GO_BIN)/gosimports
-GOMOBILE_BIN := GO111MODULE=off $(GO_BIN)/gomobile
+GOMOBILE_BIN := $(GO_BIN)/gomobile
```
Comment (guggero) [inline Makefile:13]: nit: commit message should reflect what the commit does, not the action the developer took. I know I supplied the patch, so I would've used something like "multi: bump to latest gomobile version, fix mobile build".

**Review by guggero — APPROVED** [review]
Nice, this now builds for Android on my machine.
@hsjoberg sine you probably have a dev environment for all the builds you target in Blixt, could you please take a look and make sure this works with the latest iOS versions you target?

**Comment by hsjoberg** [comment]
@guggero I can confirm iOS and macOS targets build and work with this PR.

**Comment by djkazic** [comment]
@guggero should I rebase this one too?

**Comment by djkazic** [comment]
@guggero rebased PR and branch on 0.16.1 staging branch

**Comment by Roasbeef** [comment]
Approved CI run.

**Pull request merged**